src/components/TableDialog.vue

```diff
@@ -111,130 +111,125 @@
+        fileName.value = fileName.value || getFileName(file);
+        fileUploadError.value = null;
       }
 
       const typeRecs = await csvFileTypeRecommendations(file);
-      this.columnType = Array.from(typeRecs.keys()).reduce(
+      columnType.value = Array.from(typeRecs.keys()).reduce(
         (acc, key) => ({ ...acc, [key]: typeRecs.get(key) }), {},
       );
-    },
-
-    handleUploadProgress(evt: { loaded: number; total: number; [key: string]: unknown }) {
-      this.uploadProgress = (evt.loaded / evt.total) * 100;
-    },
-
-    async createTable() {
-      const {
-        file,
-        workspace,
-        fileName,
-        key,
-        overwrite,
-      } = this;
-
-      if (file === null || fileName === null) {
+    }
+
+    // Upload options
+    const overwrite = ref(false);
+    const keyField = ref<string>(defaultKeyField);
+    const restoreKeyField = () => { keyField.value = defaultKeyField; };
+
+    // Upload state
+    const uploading = ref(false);
+    const uploadProgress = ref<number | null>(null);
+    function handleUploadProgress(evt: { loaded: number; total: number; [key: string]: unknown }) {
+      uploadProgress.value = (evt.loaded / evt.total) * 100;
+    }
+
+    // Table creation state
+    const tableDialog = ref(false);
+    const tableCreationError = ref<string | null>(null);
+    const createDisabled = computed(() => !selectedFile.value || !fileName.value || fileUploadError.value);
+    async function createTable() {
+      if (selectedFile.value === null || fileName.value === null) {
         return;
       }
 
-      this.uploading = true;
+      const { workspace } = props;
+      uploading.value = true;
 
       try {
-        await api.uploadTable(workspace, fileName, {
+        await api.uploadTable(workspace, fileName.value, {
           type: 'csv',
-          data: file,
-          key,
-          overwrite,
+          data: selectedFile.value,
+          key: keyField.value,
+          overwrite: overwrite.value,
         }, {
-          onUploadProgress: this.handleUploadProgress,
+          onUploadProgress: handleUploadProgress,
         });
 
-        this.tableCreationError = null;
-        this.tableDialog = false;
-        this.$emit('success');
+        tableCreationError.value = null;
+        tableDialog.value = false;
+
+        emit('success');
       } catch (err) {
-        this.tableCreationError = err.statusText;
+        tableCreationError.value = err.statusText;
       } finally {
-        this.uploading = false;
-        this.uploadProgress = null;
+        uploading.value = false;
+        uploadProgress.value = null;
       }
-    },
+    }
+
+    return {
+      columnType,
+      multinetTypes,
+      fileName,
+      fileUploadError,
+      tableDialog,
+      tableCreationError,
+      uploading,
+      uploadProgress,
+      createDisabled,
+      handleFileInput,
+      createTable,
+      restoreKeyField,
+      key: keyField,
```

Comment:
To try to stick with the standard practice I think this should be changed to `keyField`, and the use of `key` in the template or wherever else it appears should be updated as well.